com.unity.shadergraph/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandOverrides.cs

```diff
@@ -258,13 +258,7 @@ RenameElementCommand renameElementCommand
                 var nodeWriter = shaderGraphModel.GraphHandler.GetNodeWriter(nodeModel.graphDataName);
                 if (nodeWriter != null)
                 {
-                    var vec4Value = updatePortConstantCommand.NewValue is Vector4 value ? value : default;
-                    nodeWriter.SetPortField(portModel.UniqueName, "c0", vec4Value.x);
-                    nodeWriter.SetPortField(portModel.UniqueName, "c1", vec4Value.y);
-                    nodeWriter.SetPortField(portModel.UniqueName, "c2", vec4Value.z);
-                    nodeWriter.SetPortField(portModel.UniqueName, "c3", vec4Value.w);
-
-                    previewManager.OnLocalPropertyChanged(nodeModel.graphDataName, portModel.UniqueName, vec4Value);
+                    previewManager.OnLocalPropertyChanged(nodeModel.graphDataName, portModel.UniqueName, updatePortConstantCommand.NewValue);
```

Comment:
Inclined to feel this whole command should exclusively be triggered for the PreviewManager to call LocalPropertyChanged. The `DefaultCommandHandler` currently handles writing values from the UI into the `IConstant` which in turns reads/writes directly to CLDS.